File: Pyja-20.09.22/Pyja/20.09.22/Salak/Hepager/210606-1639 Copy XML Parama To Desktop/Global.py
# +-----------------------------------------------------------------------+
# :                                 Pyja                                  :
# :                    https://github.com/PyjaErskell                     :
# :               made by Erskell (pyja.erskell@gmail.com)                :
# :  released under the GNU AGPL v3, see <http://www.gnu.org/licenses/>.  :
# +-----------------------------------------------------------------------+

import sys

GC_MAIN = sys.modules ['__main__']

import os
import jep

def gf_jcls (x_it) : return jep .findClass (x_it)

GC_FOSA = os.sep     # (fo)lder (s)ep(a)rator
GC_PASA = os.pathsep # (pa)th (s)ep(a)rator

def gf_pj ( * x_args ) : return GC_FOSA .join (x_args) # (p)ath (j)oin
gf_xi = os.path .exists # e(xi)sts
gf_np = os.path .normpath # (n)orm(p)ath
gf_bn = os.path .basename # (b)ase(n)ame

def gp_set_os_env ( x_key, x_val ) :
  os.environ [x_key] = x_val

gp_set_os_env ( 'QT_PLUGIN_PATH', GC_MAIN .SC_QT_PLUGIN_PATH )
gp_set_os_env ( 'QT_AUTO_SCREEN_SCALE_FACTOR', '1' )

import platform
from PyQt5 import QtCore
from PyQt5.QtCore import QStandardPaths

GC_PYTHON_VR = platform .python_version ()
GC_PYQT_VR = QtCore .qVersion ()
GC_DESKTOP_PN = gf_np ( QStandardPaths .writableLocation (QStandardPaths.DesktopLocation) )

import tkinter

GC_TONO_HM = GC_MAIN .SC_TONO_HM
GC_TONO_NM = gf_bn (GC_TONO_HM)

sys.argv = [GC_TONO_NM]

GC_TCL = tkinter.Tcl ()
GC_TCL .eval ( """
  proc gf_il {x_it} { # (i)s (l)ink
    return [ expr { [file type $x_it] eq "link" } ]
  }
  proc gp_ml { x_link x_target } { # (m)ake (l)ink
    file link -symbolic $x_link $x_target
  }
  proc gp_cp { x_src x_dst { x_force 0 } } { # (c)o(p)y
    if {$x_force} {
      file copy -force $x_src $x_dst
    } else {
      file copy $x_src $x_dst
    }
  }
""")
def gf_il (x_it) :
  return GC_TCL .call ( 'gf_il', x_it ) == 1
def gp_ml ( x_src, x_link ) :
  return GC_TCL .call ( 'gp_ml', x_link, x_src )
def gp_cp ( x_src, x_dst, x_force = False ) :
  return GC_TCL .call ( 'gp_cp', x_src, x_dst, x_force )

def gf_do (x_xml_params_fn) :
  GC_MAIN .sf_do (x_xml_params_fn)

File: Pyja-19.01.22/Pyja/19.01.22/Study/JRuby/19.06.12-Rooibos/S25-Qt-Hello-Integer/src/cruby/run/Global.rb
# +-----------------------------------------------------------------------+
# :                                 Pyja                                  :
# :                    https://github.com/PyjaErskell                     :
# :               made by Erskell (pyja.erskell@gmail.com)                :
# :  released under the GNU AGPL v3, see <http://www.gnu.org/licenses/>.  :
# +-----------------------------------------------------------------------+

$GC_FOSA = File::ALT_SEPARATOR || File::SEPARATOR # (fo)lder (s)ep(a)rator
$GC_PASA = File::PATH_SEPARATOR # (pa)th (s)ep(a)rator

$GC_IS_WINDOWS = $GC_FOSA == '\\'

$gf_pj = -> ( * x_args ) { x_args .join $GC_FOSA } # (p)ath (j)oin
$gf_xi = -> (x_it) { File .exist? x_it } # e(xi)sts

$gp_set_os_env = -> ( x_key, x_val ) { ENV[x_key] = x_val }
$gf_os_env = -> (x_key) {
  fu_r = ENV[x_key]
  raise  "Can't find environment variable => #{x_key} !!!" if fu_r.nil?
  fu_r
}

$GC_KAPA_HM = $gf_pj .( 'C:', 'ProgramData', 'Bichon Frise', 'Kapa' )
$GC_JAVA_HM = $gf_pj .( $GC_KAPA_HM, '19.11.01', 'Vindue', 'x64', 'Amazon', 'Corretto', '8.232.09.1' )

$gf_bn = -> (x_fn) { File .basename x_fn }
$gf_ap = -> (x_it) {
  fu_pn = File .expand_path x_it
  return ( fu_pn .gsub '/', '\\' ) if $GC_IS_WINDOWS
  fu_pn
}

$gp_set_os_env .( 'Path', $gf_pj .( $gf_os_env .('SystemRoot'), 'System32' )  )
